<?xml version="1.0"?>
<robot name="robot">

  <!-- need but don't ues-->
  <link name="base_link"/>

<!-- mesh file fath = package://jumping_robot/mesh -->
<!-- package path = package://jumping_robot/mesh -->

  <joint name="base_to_hip" type="fixed">
    <parent link="base_link"/>
    <child link="hip_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  
  <joint name="slider_joint" type="prismatic">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="hip_link"/>
    <axis xyz="0 0 0.5"/>
    <limit effort="1000" lower="0" upper="1" velocity="100"/>
  </joint>

  <link name="hip_link">
    <inertial>
      <origin xyz="-3.7550927147833557e-10 0.004666666331953468 -0.006000000176628422" rpy="0 0 0"/>
      <mass value="0.9420000170314314"/>
      <inertia ixx="0.003243" iyy="0.000286" izz="0.003483" ixy="-0.0" iyz="6.2e-05" ixz="-0.0"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://jumping_robot/mesh/hip_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://jumping_robot/mesh/hip_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <joint name = "hip_to_thigh" type="revolute">
    <parent link="hip_link"/>
    <child link="thigh_link"/>
    <origin xyz="0 0.07 -0.03" rpy="-0.523599 0 0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-1.04719732679" upper="0.523599" velocity="10"/>
    <!-- <dynamics damping="2.0" friction="1.0"/> -->
  </joint>

  <link name="thigh_link">
    <inertial>
      <origin xyz="0.0 0.004121411947245925 -0.0921470131885182" rpy="0 0 0"/>
      <mass value="1.8213569999999997"/>
      <inertia ixx="0.005949" iyy="0.006108" izz="0.000751" ixy="-0.0" iyz="0.000491" ixz="-0.0"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://jumping_robot/mesh/thigh_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://jumping_robot/mesh/thigh_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

 <joint name = "thigh_to_shin" type="revolute">
    <parent link="thigh_link"/>
    <child link="shin_link"/>
    <origin xyz="0 0.036 -0.2" rpy="0.523599 0 0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-0.523599" upper="2.61799732679" velocity="10"/>
  </joint>

  <link name="shin_link">
    <inertial>
      <origin xyz="0.0 4.171802357206174e-18 -0.10999404803952129" rpy="0 0 0"/>
      <mass value="2.13922328936107"/>
      <inertia ixx="0.008081" iyy="0.008081" izz="0.000551" ixy="0.0" iyz="0.0" ixz="0.0"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://jumping_robot/mesh/shin_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://jumping_robot/mesh/shin_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <ros2_control name="GazeboSystem" type="system">
    <hardware>
      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
    </hardware>

    <joint name="slider_joint">
      <command_interface name="effort">
        <param name="min">-1000</param>
        <param name="max">1000</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    
    <joint name="hip_to_thigh">
      <command_interface name="position">
        <param name="min">-1.04719732679</param>
        <param name="max">0.523599</param>
      </command_interface>
      <command_interface name="effort">
        <param name="min">-1000</param>
        <param name="max">1000</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

    <joint name="thigh_to_shin">
      <command_interface name="position">
        <param name="min">-0.523599</param>
        <param name="max">2.61799732679</param>
      </command_interface>
      <command_interface name="effort">
        <param name="min">-1000</param>
        <param name="max">1000</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>

  </ros2_control>

  <gazebo>
    <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
      <robot_sim_type>gazebo_ros2_control/GazeboSystem</robot_sim_type>
      <parameters>/home/road2022/mingue/src/jumping_robot/config/controllers.yaml</parameters>
    </plugin>
  </gazebo>
  
</robot>